src/widgets/list.rs

```diff
@@ -286,7 +286,14 @@ impl<'a> StatefulWidget for List<'a> {
                 } else {
                     (x, list_area.width)
                 };
-                buf.set_line(elem_x, y + j as u16, line, max_element_width);
+                let x_offset = match line.alignment {
+                    Some(Alignment::Center) => {
+                        (area.width / 2).saturating_sub(line.width() as u16 / 2)
+                    }
+                    Some(Alignment::Right) => area.width.saturating_sub(line.width() as u16),
+                    _ => 0,
+                };
+                buf.set_line(elem_x + x_offset, y + j as u16, line, max_element_width);
```

Comment:
Do you think this could be handled by buffer::set_line so that we don't have to write this code everywhere a line is written? Not blocking, but worth checking for a future refactor.

Reply:
Yes, we can probably add an alignment parameter to the `set_line` or create `set_line_with_alignment` for refactoring. What do you think? This should be the scope of another PR though.

Reply:
Yep - definitely worth doing in another PR, but I suspect it should be the default behavior of set_line (if a line has an alignment, and we try to render the line, then it should be rendered aligned.

Reply:
Makes sense, let's create an issue for refactoring it. (probably I can work on it)